src/pages/Home.tsx

```diff
@@ -0,0 +1,204 @@
+import NavItem from "../components/NavItem";
+import Sidebar from "../components/ui/SideBar";
+import CharacterInfo from "../components/CharacterInfo";
+import CharacterInfoItem from "../components/ui/CharacterInfoItem";
+import { useQuery, gql, useReactiveVar } from "@apollo/client";
+import type { Character, PageProps } from "../utils/types";
+import { useState, useRef, useCallback } from "react";
+import { favoritesVar } from "../apollo/cache";
+
+const GET_CHARACTERS = gql`
+  query GetCharachters($page: Int) {
+    characters(page: $page) {
+      info {
+        count
+        pages
+        next
+        prev
+      }
+      results {
+        id
+        name
+        image
+        species
+        status
+        gender
+        origin {
+          name
+        }
+        location {
+          name
+        }
+        episode {
+          name
+        }
+      }
+    }
+  }
+`;
+
+export default function Home({ isMobileView, onCharacterSelect }: PageProps) {
+  const [selectedCharacter, setSelectedCharacter] = useState<Character | null>(
+    null,
+  );
+
+  const favorites = useReactiveVar(favoritesVar);
+
+  const { data, loading, error, fetchMore } = useQuery(GET_CHARACTERS, {
+    variables: { page: 1 },
+  });
+
+  const handleCharacterSelect = (id: string) => {
+    const character = data.characters.results.find(
+      (character: Character) => character.id === id,
+    );
+    setSelectedCharacter(character);
+    onCharacterSelect(character);
+  };
+
+  const handleMore = async () => {
+    await fetchMore({
+      variables: {
+        page: data.characters.info.next,
+      },
+    });
+  };
+
+  const observer = useRef<IntersectionObserver>();
+  const lastCharacterElementRef = useCallback(
+    (node: HTMLElement | null) => {
+      if (loading) return;
+      if (observer.current) observer.current.disconnect();
+
+      observer.current = new IntersectionObserver((entries) => {
+        if (entries[0].isIntersecting && data.characters.info.next) {
+          handleMore();
+        }
+      });
+
+      if (node) observer.current.observe(node);
+    },
+    [loading, data?.characters.info.next],
+  );
```

Comment:
I like you used `IntersectionObserver`. We should initialize in a useEffect hook and do a proper clean up to avoid memory leaks.